proposals/source-install.md

```diff
@@ -0,0 +1,1133 @@
+
+#### Requirements
+
+These are the requirements to build the AWS CLI v2 from source:
+
+* An environment that can run Autotools generated files (e.g.,
+  `configure`, `Makefile`). These files are widely portable across POSIX
+  platforms, and for systems that do not come with POSIX-compliant shells like
+  Windows, there is additional software available (e.g., MSYS2) that allow users
+  to run Autotools generated files. For more information, see the
+  [Appendix on Windows usage](#windows).
+
+* Python 3.8+ interpreter
+
+* (Optional) All build and runtime Python library dependencies
+  of the AWS CLI v2. This can be opted out of through configuration that
+  downloads and uses the dependencies as part of the build step.
+  See the [Downloading dependencies configuration section](#downloading-dependencies)
+  for more information.
+
+#### Configuration
+
+Configuration for the build and install of the AWS CLI v2 is specified using
+the `configure` script. For the documentation of all configuration options, run
+the `configure` script with the `--help` option:
+```bash
+./configure --help
+```
+This is a sample output from the help page:
+```
+Configures builds and installs of the AWS CLI
+
+Usage: ./configure [OPTION]... [ENV_VAR=VALUE]...
+
+Help options:
+  -h, --help              Display help
+  -V, --version           Display version
+
+Installation directories:
+  --prefix=PREFIX         Set installation prefix. By default, this value is
+                          "/usr/local".
+  --libdir=LIBDIR         Set parent directory for AWS CLI installation. The
+                          full path to the AWS CLI installation is "LIBDIR/aws-cli".
+                          The default value for "LIBDIR" is "PREFIX/lib"
+                          (i.e., "/usr/local/lib" if "--prefix" is not set).
+  --bindir=BINDIR         Set install directory for AWS CLI executables. The
+                          default value for "BINDIR" is "PREFIX/bin"
+                          (i.e., "/usr/local/bin" if "--prefix" is not set).
+
+Optional Packages:
+  --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
+  --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
+  --with-install-type=[portable-exe|system-sandbox]
+                          Specify type of AWS CLI installation. Options are:
+                          "portable-exe", "system-sandbox" (default is
+                          "frozen-exe")
+  --with-download-deps    Download all dependencies and use those when
+                          building the AWS CLI. If not specified, the
+                          dependencies (including all python packages) must be
+                          installed on your system
+
+Some influential environment variables:
+  PYTHON      the Python interpreter
+
+```
+The sections below describes the most pertinent options. 
+
+
+##### Install location
+
+Source installation of the AWS CLI v2 uses two configurable directories to
+install the AWS CLI v2:
```

Comment:
I'd like to request the addition of support for `make install DESTDIR=/path/to/dest`. ([source](https://www.gnu.org/software/make/manual/html_node/DESTDIR.html#DESTDIR))

In short, it prepends `$DESTDIR` to the `--prefix` option (and, in turn, `--libdir` and `--bindir`) exclusively for the purpose of staging files in preparation for packaging. The difference between `$DESTDIR` and simply updating `$PREFIX` is subtle, but important. I believe that this can meaningfully help with the packaging process.